tests/rust/src/bin/directory_seek.rs

```diff
@@ -27,7 +27,7 @@ unsafe fn test_directory_seek(dir_fd: wasi::Fd) {
         wasi::fd_seek(fd, 0, wasi::WHENCE_CUR)
             .expect_err("seek on a directory")
             .raw_error(),
-        wasi::ERRNO_BADF
+        wasi::ERRNO_ISDIR
```

Comment:
i guess it's more reasonable to accept both.

Reply:
I'd rather have tests very specific so implementers can use it as a guidance. IIRC @sunfishcode will look into changing that in wasmtime, do you know of any other runtime where the error code for this case is `BADF`?

Reply:
> I'd rather have tests very specific so implementers can use it as a guidance.

without having it in the spec itself? why?

> IIRC @sunfishcode will look into changing that in wasmtime, do you know of any other runtime where the error code for this case is `BADF`?

toywasm. (because it followed wasmtime)

Reply:
> without having it in the spec itself? why?

There's no error code mentioned in spec itself, so are you suggesting we should just check if there's **any** error, and ignore its value? I don't mind doing it either way, but I think even if the spec doesn't mention the value of the error code, we can still have a specific assertion here, and use it as a guidance for the future spec changes.

Reply:
my suggestion is
* accept both values here for now (not necessarily "any error". probably just BADF and ISDIR are enough.)
* start mentioning error codes in the spec

Reply:
ok, I updated it so it accepts BADF, ISDIR and NOTCAPABLE (this one is returned by WAMR).

Reply:
although NOTCAPABLE sounds very awkward to me, i guess it isn't a problem in this PR